guides/search.md

```diff
@@ -0,0 +1,289 @@
+search_body = {
+    "query": {
+        "match": {
+            "title": "dark knight"
+        }
+    },
+    "sort": [
+        {
+            "year": {
+                "order": "asc"
+            }
+        }
+    ]
+}
+# Perform the search operation on the 'movies' index with the defined query and pagination options
+response = client.search(
+    index='movies',
+    size=2,
+    from_=5,
+    body=search_body
+)
+# Extract the hits from the response
+hits = response['hits']['hits']
+
+# Print the hits
+for hit in hits:
+    print(hit)
+```
+
+With sorting, you can also use the `search_after` parameter to paginate through the search results. Let's say you have already displayed the first page of results, and you want to display the next page. You can use the `search_after` parameter to paginate through the search results. The following example will demonstrate how to get the first 3 pages of results using the search query of the previous example:
+
+```python
+# Import the required libraries
+from opensearchpy import OpenSearch
+# Create an OpenSearch client with appropriate hosts and connection details
+client = OpenSearch(hosts=['localhost'])
+# Define the search query with sorting and pagination options
+search_body = {
+    "query": {
+        "match": {
+            "title": "dark knight"
+        }
+    },
+    "sort": [
+        {
+            "year": {
+                "order": "asc"
+            }
+        }
+    ],
+    "size": 2
+}
+# Perform the search operation on the 'movies' index with the defined query and pagination options
+response = client.search(
+    index='movies',
+    body=search_body
+)
+# Extract the hits from the response
+hits = response['hits']['hits']
+# Get the last sort value from the first page
+search_after = hits[-1]['sort']
+# Fetch page 2
+search_body["search_after"] = search_after
+response = client.search(
+    index='movies',
+    body=search_body
+)
+hits_page_2 = response['hits']['hits']
+# Get the last sort value from page 2
+search_after = hits_page_2[-1]['sort']
+# Fetch page 3
+search_body["search_after"] = search_after
+response = client.search(
+    index='movies',
+    body=search_body
+)
+hits_page_3 = response['hits']['hits']
+# Print the hits from each page
+print("Page 1:")
+for hit in hits:
+    print(hit)
+print("\nPage 2:")
+for hit in hits_page_2:
+    print(hit)
+print("\nPage 3:")
+for hit in hits_page_3:
+    print(hit)
+```
+
+
+### Pagination with scroll
+
+When retrieving large amounts of non-real-time data, you can use the `scroll` parameter to paginate through the search results. 
+
+```python
+from opensearchpy import OpenSearch
+# Create an OpenSearch client with appropriate hosts and connection details
+client = OpenSearch(hosts=['localhost'])
+# Define the search query with scroll and pagination options
+search_body = {
+    "query": {
+        "match": {
+            "title": "dark knight"
+        }
+    },
+    "scroll": "1m",  # Set the scroll duration to 1 minute
```

Comment:
`scroll` is a parameter for the search request. It should be removed from here and added in L209 as follows:

```
page_1 = client.search(
    index='movies',
    scroll='1m',
    body=search_body
)
```